src/states_screens/track_info_screen.cpp

```diff
@@ -463,32 +498,57 @@ void TrackInfoScreen::eventCallback(Widget* widget, const std::string& name,
     }
     else if (name == "target-type-spinner")
     {
-        const bool timed = m_target_type_spinner->getValue() == 0;
-        UserConfigParams::m_soccer_use_time_limit = timed;
-
-        if (timed)
+        const bool target_value = m_target_type_spinner->getValue();
+        if (m_is_soccer)
         {
-            m_target_value_label->setText(_("Maximum time (min.)"), false);
-            m_target_value_spinner->setValue(UserConfigParams::m_soccer_time_limit);
+            const bool timed = target_value == 0;
+            UserConfigParams::m_soccer_use_time_limit = timed;
+
+            if (timed)
+            {
+                m_target_value_label->setText(_("Maximum time (min.)"), false);
+                m_target_value_spinner->setValue(UserConfigParams::m_soccer_time_limit);
+            }
+            else
+            {
+                m_target_value_label->setText(_("Number of goals to win"), false);
+                m_target_value_spinner->setValue(UserConfigParams::m_num_goals);
+            }
         }
-        else
+        else if (m_show_ffa_spinner)
         {
-            m_target_value_label->setText(_("Number of goals to win"), false);
-            m_target_value_spinner->setValue(UserConfigParams::m_num_goals);
+            const bool enable_ffa = target_value != 0;
+            UserConfigParams::m_use_ffa_mode = enable_ffa;
 
+            if (enable_ffa)
```

Comment:
Minor: this could be simplified into

m_target_value_label->setVisible(enable_ffa);
m_target_value_spinner->setVisible(enable_ffa);